contracts/account/AccountERC7579.sol

```diff
@@ -0,0 +1,241 @@
+    IERC1271
+{
+    using ERC7579Utils for *;
+    using EnumerableSet for *;
+
+    EnumerableSet.AddressSet private _validators;
+    EnumerableSet.AddressSet private _executors;
+    mapping(bytes4 => address) private _fallbacks;
+
+    modifier onlyModule(uint256 moduleTypeId) {
+        _checkModule(moduleTypeId, msg.sender);
+        _;
+    }
+
+    /// @inheritdoc IERC1271
+    function isValidSignature(bytes32 hash, bytes calldata signature) public view virtual override returns (bytes4) {
+        address module = abi.decode(signature[0:20], (address));
+        if (!_isModuleInstalled(MODULE_TYPE_VALIDATOR, module, msg.data)) return bytes4(0xffffffff);
+        return IERC7579Validator(module).isValidSignatureWithSender(msg.sender, hash, signature);
+    }
+
+    /// @inheritdoc IERC7579AccountConfig
+    function accountId() public view virtual returns (string memory) {
+        //vendorname.accountname.semver
+        return "@openzeppelin/contracts.erc7579account.v0-beta";
+    }
+
+    /// @inheritdoc IERC7579AccountConfig
+    function supportsExecutionMode(bytes32 encodedMode) public view virtual returns (bool) {
+        return _supportsExecutionMode(encodedMode);
+    }
+
+    /// @inheritdoc IERC7579AccountConfig
+    function supportsModule(uint256 moduleTypeId) public view virtual returns (bool) {
+        return _supportsModule(moduleTypeId);
+    }
+
+    /// @inheritdoc IERC7579Execution
+    function execute(bytes32 mode, bytes calldata executionCalldata) public virtual onlyEntryPointOrSelf {
+        _execute(Mode.wrap(mode), executionCalldata);
+    }
+
+    /// @inheritdoc IERC7579Execution
+    function executeFromExecutor(
+        bytes32 mode,
+        bytes calldata executionCalldata
+    ) public virtual onlyModule(MODULE_TYPE_EXECUTOR) returns (bytes[] memory) {
+        return _execute(Mode.wrap(mode), executionCalldata);
+    }
+
+    /// @inheritdoc IERC7579ModuleConfig
+    function isModuleInstalled(
+        uint256 moduleTypeId,
+        address module,
+        bytes calldata additionalContext
+    ) public view virtual returns (bool) {
+        return _isModuleInstalled(moduleTypeId, module, additionalContext);
+    }
+
+    /// @inheritdoc IERC7579ModuleConfig
+    function installModule(
+        uint256 moduleTypeId,
+        address module,
+        bytes memory initData
+    ) public virtual onlyEntryPointOrSelf {
+        _installModule(moduleTypeId, module, initData);
+    }
+
+    /// @inheritdoc IERC7579ModuleConfig
+    function uninstallModule(
+        uint256 moduleTypeId,
+        address module,
+        bytes memory deInitData
+    ) public virtual onlyEntryPointOrSelf {
+        _uninstallModule(moduleTypeId, module, deInitData);
+    }
+
+    function executeUserOp(
+        PackedUserOperation calldata userOp,
+        bytes32 /*userOpHash*/
+    ) public virtual override onlyEntryPointOrSelf {
+        Address.functionDelegateCall(address(this), userOp.callData[4:]);
```

Comment:
This looks weird to me. Is it specified anywhere that `executeUserOp` should behave like this? Can you provide an example?

Reply:
I would consider removing this function if it's non-standard.

Reply:
True. This is so that there are no other entrypoint in the ERC7579 account other than those defined by the standard (that are also decorated with the `withHook` modifier). I'm not sure what to do now but for now I'd rather keep the delegate call.